### ISSUE TITLE
XHR and error reports

### ISSUE THREAD

**Issue by eddow** [fuse-box/fuse-box#518]
When playing around or trying to configure heuristically, fuse-box could have more barriers to help you find a way.
There is one about the source-file loading : some source files are just present in the sourcemap after bundling but are unavailable for download.
I happen on some typos to have the browser part of Fuse downloading infinitely each 1/2s the same file (that doesn't exist)
The infinite loop is quite short and could be short-circuited and have a fool-check to stop this behaviour and display an error message
I c/p here the stack trace of the prettyfied uglyfied javascript .. it's few but it's what we have.

```
    function s(e, r) {
        if (!d)
            return r(/\.(js|json)$/.test(e) ? p.require(e) : "");
        var n = new XMLHttpRequest;
        n.onreadystatechange = function() {
            if (4 == n.readyState)
                if (200 == n.status) {
                    var i = n.getResponseHeader("Content-Type")
                      , o = n.responseText;
                    /json/.test(i) ? o = "module.exports = " + o : /javascript/.test(i) || (o = "module.exports = " + JSON.stringify(o));
                    var a = t("./", e);
                    m.dynamic(a, o),
                    r(m.import(e, {}))
                } else
                    console.error(e, "not found on request"),
                    r(void 0)
        }
        ,
        n.open("GET", e, !0),
        n.send()
    }
```
on receive, `m.import(e, {})` leads here :
```
        r.import = function(e, r) {
            return c(e, r)
        }
```
Then :
```
    function c(e, r) {
        [ ... ]
        if (!i) {
            var m = "function" == typeof r
              , x = l("async", [e, r]);
            if (x === !1)
                return;
            return s(e, function(e) {
                return m ? r(e) : null
            })
        }
        [ ... ]
    }
```

Where the call `s(e, function(e) {` closes the loop and begins a new XHR request

**Comment by nchanged** [comment]
That's a tricky one. I am sure your request returns `200OK` that's why fusebox is confused. Try setting it to `404`

**Comment by eddow** [comment]
Indeed, the infinite loop is solved in my case (and no 404 allowed in SPA, it returns the main index.html on unknown route)
This is not about how I should do not to get in an infinite loop, these "barriers" are comfort for the ones that are still uneasy with fuse and make mistakes without knowing : a simple error message can save them 1/2h of searching what might be wrong in their config (I did spend time wondering where this ame from ^^ )
Here, as it is a shallow infinite loop (3-calls deep) it may (I don't know, I didn't dig in the source) be still easy to put some fool-proofness tests in the recursion

**Comment by nchanged** [comment]
@eddow absolutely. We will solve it ;-)

**Comment by devmondo** [comment]
Fixed by #549
